pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -78,13 +80,8 @@ func (s *Reconciler) Create(ctx context.Context) error {
 		s.scope.Machine.Annotations = map[string]string{}
 	}
 
-	bootstrapToken, err := s.checkControlPlaneMachines()
-	if err != nil {
-		return errors.Wrap(err, "failed to check control plane machines in cluster")
-	}
-
 	nicName := fmt.Sprintf("%s-nic", s.scope.Machine.Name)
-	err = s.createNetworkInterface(ctx, nicName)
+	err := s.createNetworkInterface(ctx, nicName)
 	if err != nil {
```

Comment:
Just a recommendation:
```go
if err := s.createNetworkInterface(ctx, nicName); err != nil {...}
```

Though, it's still preferable so one does not test `err` variable of a different expressions. The smaller scope of a variable the better.